metadata-io/src/main/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphService.java

```diff
@@ -0,0 +1,213 @@
+import com.linkedin.metadata.graph.Edge;
+import com.linkedin.metadata.graph.GraphService;
+import com.linkedin.metadata.query.Condition;
+import com.linkedin.metadata.query.Criterion;
+import com.linkedin.metadata.query.CriterionArray;
+import com.linkedin.metadata.query.Filter;
+import com.linkedin.metadata.query.RelationshipDirection;
+import com.linkedin.metadata.query.RelationshipFilter;
+import com.linkedin.metadata.search.elasticsearch.indexbuilder.SettingsBuilder;
+import com.linkedin.metadata.utils.elasticsearch.IndexConvention;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.GetIndexRequest;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ElasticSearchGraphService implements GraphService {
+
+  private static final int MAX_ELASTIC_RESULT = 10000;
+  private final RestHighLevelClient searchClient;
+  private final IndexConvention _indexConvention;
+  private final ESGraphWriteDAO _graphWriteDAO;
+  private final ESGraphQueryDAO _graphReadDAO;
+
+  private static final String DOC_DELIMETER = "--";
+  public static final String INDEX_NAME = "graph_service_v1";
+  private static final Map<String, Object> EMPTY_HASH = new HashMap<>();
+
+  private String toDocument(@Nonnull final Edge edge) {
+    final ObjectNode searchDocument = JsonNodeFactory.instance.objectNode();
+
+    final ObjectNode sourceObject = JsonNodeFactory.instance.objectNode();
+    sourceObject.put("urn", edge.getSource().toString());
+    sourceObject.put("entityType", edge.getSource().getEntityType());
+
+    final ObjectNode destinationObject = JsonNodeFactory.instance.objectNode();
+    destinationObject.put("urn", edge.getDestination().toString());
+    destinationObject.put("entityType", edge.getDestination().getEntityType());
+
+    searchDocument.set("source", sourceObject);
+
+    searchDocument.set("destination", destinationObject);
+
+    searchDocument.put("relationshipType", edge.getRelationshipType());
+
+    return searchDocument.toString();
+  }
+
+  private String toDocId(@Nonnull final Edge edge) {
+    String rawDocId =
+        edge.getSource().toString() + DOC_DELIMETER + edge.getRelationshipType() + DOC_DELIMETER + edge.getDestination().toString();
+
+    try {
+      byte[] bytesOfRawDocID = rawDocId.getBytes("UTF-8");
+      MessageDigest md = MessageDigest.getInstance("MD5");
```

Comment:
Any reason you went with this for the doc id? just curious!

Reply:
yup- elastic limits doc id length and we have hit errors even in search where the doc_id is too long